Use proper names for Calico certificates and remove unnecessary ceritifcates

r7vme opened the pull request:
Part of: https://github.com/giantswarm/giantswarm/issues/4173

Naming aligned with Calico CNI script https://github.com/projectcalico/cni-plugin/blob/release-v3.2/k8s-install/scripts/install-cni.sh#L40-L55

xh3b4sd reviewed (approved):
When you have the migration in mind then lets go with this.

kopiczko commented on files.go:
```diff
@@ -72,19 +72,6 @@ func newFilesClusterCommon(cluster Cluster) Files {
-		},
-		{
-			AbsolutePath: "/etc/kubernetes/ssl/etcd/calico-client-key.pem",
-			Data:         cluster.CalicoEtcdClient.Key,
```
Can we keep `CalicoEtcdClient` and remove `CalicoClient` instead? I think it's clearer to have `Etcd` in the name.

Commit:
Create proper calico certificates

r7vme commented:
PTAL, addressed comment.

Migration is easy because we don't use Calico certificates now. Anyway all tracked here https://github.com/giantswarm/giantswarm/issues/4173

tuommaki reviewed (approved):
LGTM, but might be worth to properly test (as last time I modified this, there were issues later that hit the fan).

r7vme commented:
> Can we just simply remove the certs there?

Yes.

So the calico certs are no longer needed?

We had two copies of calico certs and none were used. This change just removes one and changes to proper filenames for another.

So in short, they are not used, so yes can be removed.

> What about the certs being there? Will they be cleaned up somehow?

Seems not. Do you have an idea how it can be done?

xh3b4sd commented:
> Seems not. Do you have an idea how it can be done?

This is part of a migration plan we should have. We could just add some goroutine to `cluster-operator` checking if these things are there and if so delete them. Quick and dirty. Deploy. Wait. Cleanup code.

xh3b4sd commented:
I remove the calico certs from `kubernetesd` then.

xh3b4sd commented:
I commented here that we should have a TODO issue. Where is it? This is supposed to be tracked in the code with the TODO comment so we have a cross link and do not have to search.
